app/lib/submission_builder/base.rb

```diff
@@ -0,0 +1,38 @@
+module SubmissionBuilder
+  class Base
+    include SubmissionBuilder::FormattingMethods
+    attr_accessor :submission
+    class << self
+      attr_reader :schema_file, :root_node
+    end
+
+    @schema_file = nil # class instance variable
+
+    def initialize(submission, validate: true)
+      @submission = submission
+      @validate = validate
+    end
+
+    def root_node_attrs
+      { "xmlns:efil" => "http://www.irs.gov/efile", "xmlns" => "http://www.irs.gov/efile" }
+    end
+
+    def document
+      raise "SubmissionBuilder classes must implement their own document method that returns a Nokogiri::XML::Document object"
+    end
+
+    def build
+      errors = []
+      if @validate
```

Comment:
The switch for validating against the schema allows us the option to validate each "built" piece against its own schema, but will allow us to bypass individual validation on the final "Return" object so that we can validate only at the top level and get all relevant xml errors in one response.